### DISCUSSION TITLE
Help：Question about the Communication protocols and security between flux components

### DISCUSSION THREAD

**Discussion by wangyuan249** [fluxcd/flux2#4280] [Q&A]
Hi , I found that the communication between flux components currently is based on HTTP. 
Such as  kustomize-controller -> source-controller get artifact info through "http://source-controller.flux-system.svc.cluster.local.:xxxx" )
![image](https://github.com/fluxcd/flux2/assets/35907099/ea28e187-5b1a-40cb-a25c-667a06b45f62)
![image](https://github.com/fluxcd/flux2/assets/35907099/fe747ef5-0e28-4442-abb1-1d82c5970db0)

or event reporting through：
![image](https://github.com/fluxcd/flux2/assets/35907099/a444ad7f-5385-4559-b9e8-65e120d02669)

Is it possible to modify the configuration or switch it to HTTPS?

**Comment by makkes** [comment]
This is currently not entirely possible and we're working on an RFC to change this situation: https://github.com/fluxcd/flux2/pull/3368/

**Reply by wangyuan249** [reply]
Got it, looking forward to the TLS encrypted version.
